src/azul/indexer/document.py

```diff
@@ -512,18 +519,22 @@ def to_index(self, catalog: Optional[CatalogName], field_types: CataloguedFieldT
             '_id' if bulk else 'id': self.coordinates.document_id
         }
         if self.version_type is VersionType.none:
-            assert self.version is None
             if bulk:
                 result['_op_type'] = 'delete' if delete else 'index'
             else:
                 # For non-bulk updates, the op-type is determined by which
                 # client method is invoked.
-                pass
+                if delete:
+                    result['if_seq_no'], result['if_primary_term'] = self.version
```

Comment:
See, here `self.version_type` is `VersionType.none` but you're extracting a version. That's very counter intuitive and doesn't quite add up.